boards/arm/96b_neonkey/doc/96b_neonkey.rst

```diff
@@ -0,0 +1,227 @@
+Default settings are 115200 8N1.
+
+I2C
+---
+
+96Boards Neonkey board has up to 3 I2Cs. The default I2C mapping for Zephyr is:
+
+- I2C1_SCL : PB6
+- I2C1_SDA : PB7
+- I2C2_SCL : PB10
+- I2C2_SDA : PB3
+- I2C3_SCL : PA8
+- I2C3_SCL : PB4
+
+Programming and Debugging
+*************************
+
+Building
+========
+
+Build the Zephyr kernel and application with:
+
+.. code-block:: console
+
+   $ cd <zephyr_root_path>
+   $ source zephyr-env.sh
+   $ cd $ZEPHYR_BASE/samples/hello_world/
+   $ make BOARD=96b_neonkey
+
+Flashing
+========
+
+96Boards Neonkey can be flashed by two methods, one using the ROM
+bootloader and another using the SWD debug port (which requires additional
+hardware).
+
+Using ROM bootloader:
+---------------------
+
+ROM bootloader can be triggered by the following pattern:
+
+1. Connect BOOT0 to VDD (link JTAG pins 1 and 5 on P4 header)
+2. Hold the USR button
+3. Press the RST button
+
+More detailed information on activating the ROM bootloder can be found in
+Chapter 29 of Application note `AN2606`_. The ROM bootloader supports flashing
+via UART, I2C and SPI protocols.
+
+For flashing, `stm32flash`_ command line utility can be used. The following
+command will flash the ``zephyr.bin`` binary to the Neonkey board using UART
+and starts its execution:
+
+.. code-block:: console
+
+   $ stm32flash -w zephyr.bin -v -g 0x08000000 /dev/ttyS0
+
+.. note::
+   The above command assumes that Neonkey board is connected to
+   serial port ``/dev/ttyS0``.
+
+Using SWD debugger:
+-------------------
+
+For flashing via SWD debug port, 0.1" male header must be soldered at P4
+header available at the bottom of the board, near RST button.
+
+Use the `Black Magic Debug Probe`_ as an SWD programmer, which can
+be connected to the P4 header using its flying leads and its 20 Pin
+JTAG Adapter Board Kit. When plugged into your host PC, the Black
+Magic Debug Probe enumerates as a USB serial device as documented on
+its `Getting started page`_.
+
+It also uses the GDB binary provided with the Zephyr SDK,
+``arm-zephyr-eabi-gdb``. Other GDB binaries, such as the GDB from GCC
+ARM Embedded, can be used as well.
+
+.. code-block:: console
+
+   $ arm-zephyr-eabi-gdb -q zephyr.elf
+   (gdb) target extended-remote /dev/ttyACM0
+   Remote debugging using /dev/ttyACM0
+   (gdb) monitor swdp_scan
+   Target voltage: 1.8V
+   Available Targets:
+   No. Att Driver
+    1      STM32F4xx
+   (gdb) attach 1
+   Attaching to Remote target
+   0x080005d0 in ?? ()
+   (gdb) load
+
+Debugging
+=========
+
+After flashing 96Boards Neonkey, it can be debugged using the same
+GDB instance. To reattach, just follow the same steps above, till
+"attach 1". You can then debug as usual with GDB. In particular, type
+"run" at the GDB prompt to restart the program you've flashed.
+
+References
+**********
+
+.. _96Boards website:
+   https://www.96boards.org/product/neonkey/
+
+.. _AN2606:
+   www.st.com/resource/en/application_note/cd00167594.pdf
```

Comment:
needs the https:// protocol added (otherwise a relative link is generated)

Reply:
Done